src/modules/hooks/useBlockExplorer/useBlockExplorer.ts

```diff
@@ -24,29 +25,46 @@ export interface IBuildEntityUrlParams {
      * The type of the entity (e.g. address, transaction, token)
      */
     type: ChainEntityType;
+    /**
+     * ID of the chain related to the entity. When set, overrides the chainId set as hook parameter.
+     */
+    chainId?: number;
     /**
      * The ID of the entity (e.g. transaction hash for a transaction)
      */
     id?: string;
 }
 
 export const useBlockExplorer = (params?: IUseBlockExplorerParams) => {
-    const { chains, chainId } = params ?? {};
+    const { chains, chainId: hookChainId } = params ?? {};
 
     const globalChains = useChains();
-    const processedChains = chains ?? globalChains;
 
-    const chainDefinitions = chainId ? processedChains.find((chain) => chain.id === chainId) : processedChains[0];
-    const { default: blockExplorer } = chainDefinitions?.blockExplorers ?? {};
+    const getBlockExplorer = useCallback(
+        (chainId?: number) => {
+            const processedChains = chains ?? globalChains;
+            const processedChainId = chainId ?? hookChainId;
+
+            const chainDefinitions = processedChainId
+                ? processedChains.find((chain) => chain.id === processedChainId)
+                : processedChains[0];
+
+            return chainDefinitions?.blockExplorers?.default;
+        },
+        [chains, globalChains, hookChainId],
+    );
 
     const buildEntityUrl = useCallback(
-        ({ type, id }: IBuildEntityUrlParams) => {
+        ({ type, chainId, id }: IBuildEntityUrlParams) => {
+            const blockExplorer = getBlockExplorer(chainId ?? hookChainId);
             const baseUrl = blockExplorer?.url;
 
             return baseUrl != null ? `${baseUrl}/${type}/${id}` : undefined;
         },
-        [blockExplorer],
+        [getBlockExplorer, hookChainId],
     );
 
-    return { blockExplorer, buildEntityUrl };
+    const blockExplorer = getBlockExplorer(hookChainId);
+
+    return { blockExplorer, getBlockExplorer, buildEntityUrl };
```

Comment:
how do we feel about this naming generally? I feel like it's all kind of stepping on it own different parts in terms of clarity? 

- useBlockExplorer
   - getBlockExplorerDefaultDefinition
   - getBlockExplorerLinkByType
   - getBlockExplorerDefaultByChainId
   
thoughts?

Reply:
Mm.. I wouldn't be that specific on the result vars, `getBlockExplorerDefaultByChainId` feels like the `chainId` parameter is required while it's optional and fallsback to the one set on the hook parameter. Also wouldn't rename `blockExplorer` to `getBlockExplorerDefaultDefinition`, maybe `blockExplorerDefinition`? But both works fine

Reply:
Hmmm yeah I mean it's kind of overkill generally, like IYKYK, but their distinctions just seemed a little muddy to me, even looking at it from the test suite perspective. buildEntityLink is the most distinct and yet I wouldn't really know what that means without seeing docs. It's almost more like hash type? `generateResourceLink` `blockExplorerDefiniton` `getBlockExplorerDetails`